src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -26,6 +26,7 @@ private sealed class EasyRequest : TaskCompletionSource<HttpResponseMessage>
             internal readonly HttpRequestMessage _requestMessage;
             internal readonly CurlResponseMessage _responseMessage;
             internal readonly CancellationToken _cancellationToken;
+            internal readonly ClientCertProvider _clientCertProvider;
```

Comment:
The only reason we're storing this here is to be able to dispose it, right?  Otherwise it could be a local in the SetClientCertificateOptions function, which could do the Create call and keep everything local?  Seems like it'll actually end up being relatively rare that we have one of these, only happening if you're connecting to an https URL and you're using Automatic for certs and you're using a libcurl with an ssl implementation that doesn't support the callback and you have client certs.  Given that, it's unfortunate to increase the size of EasyRequest for all requests.  Since it's just disposal that'd be needed, instead of having this field, only in the case where you need the disposal you could register a continuation from this.Task, and have that continuation do the disposal, e.g.
```C#
this.Task.ContinueWith((_, state) => ((IDisposable)state).Dispose(), provider, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
```
That'd make it pay-for-play.